extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/runtime/ValidationSupport.java

```diff
@@ -0,0 +1,96 @@
+    public static final String PROBLEM_FIELD = "field";
+    public static final String PROBLEM_MESSAGE = "message";
+    public static final String PROBLEM_STATUS = "status";
+    public static final String PROBLEM_VIOLATIONS = "violations";
+
+    private ValidationSupport() {
+        // Avoid direct instantiation
+    }
+
+    public static Validator getValidator(ArcContainer container) {
+        return container.instance(Validator.class).get();
+    }
+
+    public static String mapViolationsToJson(Set<ConstraintViolation<?>> violations, HttpServerResponse response) {
+        response.setStatusCode(500);
+        JsonObject json = generateJsonResponse(violations, true);
+        return json.encode();
+    }
+
+    /**
+     * Generates a JSON response following https://opensource.zalando.com/problem/constraint-violation/
+     * 
+     * @param violations the violations
+     * @return the json object
+     */
+    private static JsonObject generateJsonResponse(Set<ConstraintViolation<?>> violations, boolean violationInProducedItem) {
+        JsonObject json = new JsonObject()
+                .put(PROBLEM_TITLE, "Constraint Violation")
+                .put(PROBLEM_DETAIL, "validation constraint violations");
+
+        JsonArray array = new JsonArray();
+        boolean isProduced = false;
+        for (ConstraintViolation<?> cv : violations) {
+            if (cv.getExecutableReturnValue() != null) {
+                isProduced = true;
+            }
+            JsonObject violation = new JsonObject();
+            violation.put(PROBLEM_FIELD, cv.getPropertyPath().toString());
+            violation.put(PROBLEM_MESSAGE, cv.getMessage());
+            array.add(violation);
+        }
+        json.put(PROBLEM_STATUS, isProduced || violationInProducedItem ? 500 : 400);
+        json.put(PROBLEM_VIOLATIONS, array);
+        return json;
+    }
+
+    public static void handleViolationException(ConstraintViolationException ex, RoutingContext rc) {
+        String accept = rc.request().getHeader(ACCEPT_HEADER);
+        if (accept != null && accept.contains(APPLICATION_JSON)) {
+            rc.response().putHeader(RouteHandlers.CONTENT_TYPE, APPLICATION_JSON);
+            JsonObject json = generateJsonResponse(ex.getConstraintViolations(), false);
+            rc.response().setStatusCode(json.getInteger(PROBLEM_STATUS));
+            rc.response().end(json.encode());
+        } else {
+            // Check status
+            int status = 400;
+            for (ConstraintViolation<?> constraintViolation : ex.getConstraintViolations()) {
+                if (constraintViolation.getExecutableReturnValue() != null) {
+                    status = 500;
+                    break;
+                }
+            }
+            // If not JSON just fails.
+            rc.fail(status, ex);
+        }
+    }
+
+    public static void log(String s) {
+        System.out.println("[LOG] - " + s);
```

Comment:
Hmmm, should we really use `System.out.println` here and below?

Reply:
Are the log methods even used somewhere?

Reply:
unused, was only there for debugging

Reply:
Removed